src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,438 @@
+        } catch (err) {
+            core.error(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Using a Dockerfile that was provided or found at the root of the application source,
+     * creates a runable application image.
+     * @param imageToDeploy - the name of the runnable application image that is created and can be later deployed
+     * @param appSourcePath - the path to the application source on the machine
+     * @param dockerfilePath - the path to the Dockerfile to build and tag with the provided image name
+     */
+    public async createRunnableAppImageFromDockerfile(
+        imageToDeploy: string,
+        appSourcePath: string,
+        dockerfilePath: string) {
+        core.debug(`Attempting to create a runnable application image from the provided/found Dockerfile "${dockerfilePath}" with image name "${imageToDeploy}"`);
+        try {
+            const dockerTool = await io.which("docker", true);
+            await new Utility().executeAndthrowIfError(dockerTool, ['build', '--file', `${dockerfilePath}`, `${appSourcePath}`, '--tag', `${imageToDeploy}`]);
+            core.debug(`Successfully created runnable application image from the provided/found Dockerfile "${dockerfilePath}" with image name "${imageToDeploy}"`);
+        } catch (err) {
+            core.setFailed(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Determines the runtime stack to use for the runnable application image.
+     * @param appSourcePath - the path to the application source on the machine
+     * @returns a string representing the runtime stack that can be used for the Oryx MCR runtime images
+     */
+    public async determineRuntimeStackAsync(appSourcePath: string): Promise<string> {
+        core.debug('Attempting to determine the runtime stack needed for the provided application source');
+        try {
+            const dockerTool: string = await io.which("docker", true);
+            // Use 'oryx dockerfile' command to determine the runtime stack to use and write it to a temp file
+            await new Utility().executeAndthrowIfError(dockerTool, ['run', '--rm', '-v', `${appSourcePath}:/app`, `${ORYX_CLI_IMAGE}`, '/bin/bash', '-c', `oryx dockerfile /app | head -n 1 | sed 's/ARG RUNTIME=//' >> /app/oryx-runtime.txt`])
+
+            // Read the temp file to get the runtime stack into a variable
+            const oryxRuntimeTxtPath = path.join(appSourcePath, 'oryx-runtime.txt');
+            let command: string = `head -n 1 ${oryxRuntimeTxtPath}`;
+            if (IS_WINDOWS_AGENT) {
+                command = `Get-Content -Path ${oryxRuntimeTxtPath} -Head 1`;
+            }
+
+            const runtimeStack = await new CommandHelper().execCommandAsync(command);
+
+            // Delete the temp file
+            command = `rm ${oryxRuntimeTxtPath}`;
+            if (IS_WINDOWS_AGENT) {
+                command = `Remove-Item -Path ${oryxRuntimeTxtPath}`;
+            }
```

Comment:
As mentioned previously, we shouldn't need to differentiate between OS used when running commands, but we _should_ keep the distinction in this file when determining which URL to use when downloading the pack CLI (which is done later in this file)

Reply:
Updated.